axelor-account/src/main/java/com/axelor/apps/account/web/AccountChartController.java

```diff
@@ -58,4 +68,259 @@ public void installChart(ActionRequest request, ActionResponse response) throws
 
     } else response.setFlash(I18n.get(IExceptionMessage.ACCOUNT_CHART_3));
   }
+
+  public void chartInvoicedTurnoverThisYearVsLastyear(
+      ActionRequest request, ActionResponse response) {
+    List<Map<String, Object>> dataList = new ArrayList<>();
+    InvoiceRepository repo = Beans.get(InvoiceRepository.class);
+    BigDecimal salesThisYear =
+        repo.all()
+            .filter(
+                "self.statusSelect != 1 and self.operationTypeSelect=3 and DATE_PART('year', self.invoiceDate) = DATE_PART('year',CURRENT_DATE) ")
+            .fetchStream()
+            .map(i -> i.getExTaxTotal())
+            .reduce(BigDecimal.ZERO, BigDecimal::add);
+    BigDecimal salesLastYear =
+        repo.all()
```

Comment:
Create a JPQL query using SUM instead of computing the sum in java

Reply:
used SQL query instead for every charts